Bunch of (not only) s390x fixes [v2]

ldoktor opened the pull request:
This is a follow-up of the https://github.com/autotest/tp-qemu/pull/1163 which got some acks and some nacks.

Changes:
* block_copy: Improve block regexp to work with recent qemu -> improved regexp to support spaces in filename
* Make sure images are removed in live_snapshot_chain -> reworked to not hide exceptions from test execution
* Turn the iscsi target into example only -> instead of commenting-out keep it and change the values to obviously dummy ones along with note to change it downstream
* Adjust config of nic_hotplug to work properly on s390x -> removed the double-fix

All but the changed commits were already acked by @luckyh so waiting for 2nd ack there and full 2 acks are required for the 4 patches mentioned above.

clebergnu commented on qemu/tests/block_copy.py:
```diff
@@ -264,9 +264,21 @@ def get_image_file(self):
+                    if image_file[0][-1] == ' ':
+                        return image_file[0][:-1]
+                    else:
+                        return image_file[0]
```
This fixes errors such as:

```
Reproduced traceback from: /home/cleber/src/avocado/avocado-vt/avocado_vt/test.py:407
Traceback (most recent call last):
  File "/home/cleber/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/block_commit_stress.py", line 89, in run
    stress_test = BlockCommitStress(test, params, env, tag)
  File "/home/cleber/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/block_copy.py", line 48, in __init__
    self.image_file = self.get_image_file()
  File "/home/cleber/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/block_copy.py", line 268, in get_image_file
    return image_file[0]
IndexError: list index out of range
```

When running a test such as `type_specific.io-github-autotest-qemu.block_commit.with_payload.with_stress`.

ACK.

clebergnu commented on qemu/tests/cfg/live_snapshot_chain.cfg:
```diff
@@ -38,6 +38,8 @@
             file_create = yes
+            # Check base, middle and last (set in code) images
             check_base_image_image1 = yes
+            check_base_image_sn32 = yes
```
ACK.

clebergnu commented on qemu/tests/cgroup.py:
```diff
@@ -602,8 +604,8 @@ def _test(direction, blkio):
         stat_cmd = "killall -SIGUSR1 dd; true"
-        re_dd = (r'(\d+) bytes \(\d+\.*\d* \w*\) copied, (\d+\.*\d*) s, '
-                 '\d+\.*\d* \w./s')
+        re_dd = (r'(\d+) bytes \(\d+\.*\d* \w*(, \d+\.*\d* \w*)?\) copied, '
```
I couldn't properly follow this change (unless I spent too much time running and debugging this code).  Can you explain why this changed with the `shell=True` change?

ldoktor replied:
I just forgot to separate this change. Newer `dd` versions have different format. Do you want me to split these patches or can it be part of it?

ldoktor replied:
OK, I'm splitting in v3 and omitting in the push

clebergnu commented on qemu/tests/cgroup.py:
```diff
@@ -1668,8 +1668,8 @@ def _get_stat(pid):
                     out = process.system_output("cat /proc/%s/task/*/stat" %
-                except exceptions.CmdError:
                                                 pid, shell=True)
+                except process.CmdError:
```
ACK.

clebergnu commented on qemu/tests/cfg/unittest_kvmctl.cfg:
```diff
@@ -1,5 +1,6 @@
 - unit_test_kvmctl:
     no JeOS
+    only i386, x86_64
```
ACK.

clebergnu commented on qemu/tests/time_manage.py:
```diff
@@ -86,8 +86,7 @@ def run(test, params, env):
                 # Get the respective vm object
-                vmname = "virt-tests-vm%d" % (vmid + 1)
-                vm = env.get_vm(vmname)
+                vm = env.get_vm(vmnames[vmid])
```
ACK.

clebergnu commented on qemu/tests/cfg/virtio_scsi_mq.cfg:
```diff
@@ -9,5 +9,5 @@
     q35:
-        system_image_drive_format = ahci
         dev_type = q35-pcihost
+    system_image_drive_format = virtio_blk
```
Since this is a RHEL only test, ACK.

clebergnu commented on qemu/tests/cfg/virtio_scsi_mq.cfg:
```diff
@@ -8,6 +8,4 @@
     start_vm = no
     image_size_extra_images = 512M
-    q35:
-        dev_type = q35-pcihost
```
ACK.

clebergnu commented on qemu/tests/cfg/boot_from_device.cfg:
```diff
@@ -76,9 +76,10 @@
+            # Please update these to point to your iscsi server
+            portal_ip = 127.0.0.1
+            initiator = "iqn.2010-07.com.example:my_initiator"
+            target = "iqn.2001-05.com.mytarget:0-0a0000-000000000-0000000000000000-my-target
```
This is missing a closing double-quote.

ldoktor replied:
OK, fixing in v3 as it lacks 2 acks

clebergnu commented on qemu/tests/cfg/nic_hotplug.cfg:
```diff
@@ -21,7 +21,7 @@
                 pci_model = virtio-net-ccw
         - nic_e1000:
-            no ppc64 ppc64le
+            no ppc64, ppc64le, s390x
```
ACK.

clebergnu commented on qemu/tests/cfg/floppy_test.cfg:
```diff
@@ -1,7 +1,6 @@
     type = floppy
-    no pseries
-    no q35
+    only i440fx
```
ACK.

clebergnu commented on qemu/tests/cfg/live_backup.cfg:
```diff
@@ -24,7 +24,7 @@
             Linux:
                 create_cmd = "dd if=/dev/urandom of=FILE bs=128k count=100"
-                file_names = "/home/test"
+                file_names = "/home/test.img"
```
ACK.

clebergnu commented on qemu/tests/cfg/seabios.cfg:
```diff
@@ -1,6 +1,6 @@
 - seabios: install setup image_copy unattended_install.cdrom
     no Host_RHEL.m5
-    no ppc64 ppc64le aarch64
+    only i386, x86_64
```
ACK.

clebergnu commented on qemu/tests/cfg/qemu_guest_agent.cfg:
```diff
@@ -134,8 +134,8 @@
-            # ppc64 and ppc64le platform doesn't support isa serial
-            no ppc64 ppc64le
+            # isa-serial is only supported on x86
+            only i386, x86_64
```
ACK.

clebergnu commented on qemu/tests/cfg/balloon_check.cfg:
```diff
@@ -73,7 +73,7 @@
             balloon_type_enlarge = enlarge
         - balloon_fix_value:
-            no ppc64 aarch64
+            only i386, x86_64
```
ACK.

clebergnu commented on qemu/tests/cfg/multi_disk.cfg:
```diff
@@ -48,6 +48,7 @@
         - all_drive_format_types:
             # virtio-scsi driver support from RHEL6.3
             no RHEL.3 RHEL.4 RHEL.5 RHEL.6.0 RHEL.6.1 RHEL.6.2
+            no s390x
```
ACK.

clebergnu commented on qemu/tests/cfg/balloon_hotplug.cfg:
```diff
@@ -16,6 +16,9 @@
     balloon_type_enlarge = enlarge
+    balloon_device = virtio-balloon-pci
+    s390x:
+        balloon_device = virtio-balloon-ccw
```
Minor typo in the commit message: `s/ballon/balloon/`

ldoktor replied:
Fixing and updating here in v2

clebergnu commented on qemu/tests/ksm_overcommit.py:
```diff
@@ -431,11 +433,14 @@ def split_parallel():
-        host_reserve = ((utils_memory.memtotal() -
-                         utils_memory.read_from_meminfo("MemFree")) /
-                        1024 + 128)
+        host_reserve = ((utils_memory.memtotal() - available) / 1024 + 128)
```
Oh yeah.  How long before we get to live in a "drop caches" free world? :laughing:

ACK.

clebergnu commented on qemu/tests/cfg/spice.cfg:
```diff
@@ -13,11 +13,11 @@
+        - RHEL-6-devel-i386:
             image_name_vm2 = images/rhel6devel-32_client
-        -RHEL.7.devel.x86_64:
+        - RHEL-7-devel-x86_64:
```
ACK.

clebergnu commented:
@ldoktor found a couple (or so) of very small issues.  Also, please consider taking a look at #1173 while it's hot, as it's closely related to these changes here (and built on top of them).

ldoktor commented on qemu/tests/live_snapshot_chain.py:
```diff
@@ -79,97 +98,99 @@ def do_operate(params, key_word):
+                      logging.error)
+        if vm.is_alive():
+            vm.destroy()
+        cleanup_images(snapshot_chain, params)
```
No acks here, skipping in push-request.

ldoktor commented:
Changes.

* removed "Make sure images are removed in live_snapshot_chain" (no acks)
* removed "Fix the usage of process in cgroup test" (probably needs splitting, waiting for @clebergnu  answer)
* removed "Turn the iscsi target into example only" (needs 2nd ack and minor fix)
* modified commit message of "Use correct device in ballon_hotplug test" (use balloon)

The `removed` commits will be sent as v3.
